plutus-core-spec/figures/TypedCekMachine.tex

```diff
@@ -0,0 +1,74 @@
+\documentclass[../plutus-core-specification.tex]{subfiles}
+
+\begin{document}
+
+\begin{figure}[!ht]
+\begin{subfigure}[c]{\linewidth}        %%% ---------------- CEK machine states ---------------- %%%
+\[
+\begin{array}{lrcl}
+        \textrm{Stack} & s      & ::= & \cdot \enspace | \enspace s,f    \\
+        \blue{ \textrm{Environment}} & \rho & ::= & [] \enspace | \enspace \rho[x \mapsto C] \\
+        \blue{ \textrm{Closure}} & C  & ::= & (V,\rho) \\
+        \textrm{State} & \sigma & ::= & s;\rho \compute M \enspace | \enspace s;\rho \return V  \enspace | \enspace \ckerror{} \enspace | \enspace \square (V,\rho)
+    \end{array}
+\]
+\caption{States}
+\end{subfigure}
+
+\vspace{1mm}
+\hrule
+\vspace{2mm}
+
+\begin{subfigure}[c]{\linewidth}  %%% ---------------- CEK machine frames ---------------- %%%
+\[
+    \begin{array}{rlr}
+       f ::= & \inInstLeftFrame{A}                     & \textrm{left instantiation}\\
+             & \inWrapRightFrame{A}{B}            & \textrm{right wrap}\\
+             & \inUnwrapFrame{}                        & \textrm{unwrap}\\
+             & \inAppLeftFrame{(M,\rho)}                 & \textrm{left application}\\
+             & \inAppRightFrame{C}                & \textrm{right application}\\
```

Comment:
Looks like this should also have a closure?

Reply:
You mean it should have a C where it says (M, \rho)?  That's a bit annoying.  I thought I could get away with defining closures to be of the form (V,\rho) with V a _value_.  That's what we want everywhere else, but in [M N] we unfortunately need to save the environment for M while we're evaluating N, so I was attempting to do that without actually saying we have a closure.

Reply:
Oh I see, C is a closure. I actually thought we were missing the closure in the right-frame case.

The machine just specifies both as a pair of an environment and a term/value respectively, not as a closure, I guess for this reason (@effectfully ?).

Reply:
> The machine just specifies both as a pair of an environment and a term/value respectively, not as a closure, I guess for this reason (@effectfully ?).

No, it's a positive side effect of me not trying to come up with any abstraction there.